sql/moz-fx-data-marketing-prod/ga_derived/www_site_downloads_v2/query.sql

```diff
@@ -1,43 +1,97 @@
-SELECT
-  PARSE_DATE('%Y%m%d', event_date) AS date,
-  a.user_pseudo_id || '-' || CAST(e.value.int_value AS string) AS visit_identifier,
-  device.category AS device_category,
-  device.operating_system AS operating_system,
-  device.language AS `language`,
-  geo.country AS country,
-  collected_traffic_source.manual_source AS source,
-  collected_traffic_source.manual_medium AS medium,
-  collected_traffic_source.manual_campaign_name AS campaign,
-  collected_traffic_source.manual_content AS ad_content,
-  device.web_info.browser AS browser,
+WITH staging AS (
+  SELECT
+    `date`,
+    visit_identifier,
+    device_category,
+    operating_system,
+    `language`,
+    country,
+    source,
+    medium,
+    campaign,
+    ad_content,
+    browser,
   --note: the 2 columns are the same because in GA4, there is no logic saying you can only count 1 download per session, unlike GA3
-  COUNT(1) AS download_events,
-  COUNT(1) AS downloads,
-  COUNTIF(
-    NOT `moz-fx-data-shared-prod.udf.ga_is_mozilla_browser`(device.web_info.browser)
-  ) AS non_fx_downloads,
-FROM
-  `moz-fx-data-marketing-prod.analytics_313696158.events_*` AS a
-JOIN
-  UNNEST(event_params) AS e
-WHERE
-  _TABLE_SUFFIX = FORMAT_DATE('%Y%m%d', @submission_date)
-  AND e.key = 'ga_session_id'
-  AND e.value.int_value IS NOT NULL
-  AND (
-    (a.event_name = 'product_download' AND a.event_date <= '20240216')
-    OR (
-      a.event_name IN (
-        'firefox_download',
-        'focus_download',
-        'klar_download',
-        'firefox_mobile_download'
+    COUNTIF(
+    --prior to and including 2/16/24
+      (
+        `date` <= '2024-02-16'
+        AND event_name = 'product_download'
+        AND platform_type IN (
+          'win',
+          'win64',
+          'macos',
+          'linux64',
+          'win64-msi',
+          'linux',
+          'win-msi',
+          'win64-aarch64'
+        )
+        AND product_type = 'firefox'
       )
-      AND a.event_date > '20240216'
-    )
-  )
-GROUP BY
-  date,
+      OR
+    --on and after 2/17/24
+      (`date` >= '2024-02-17' AND event_name = 'firefox_download')
+    ) AS download_events
+  FROM
+    (
+      SELECT
```

Comment:
nit: I'd suggest using a CTE instead of a subquery, generally find it's a lot easier to read the code.

Reply:
Updated